src/util/mod.rs

```diff
@@ -420,6 +422,36 @@ pub fn monitor_threads<S: Into<String>>(_: S) -> io::Result<()> {
     Ok(())
 }
 
+/// A simple ring queue with fixed capacity.
```

Comment:
There are two implementations out there. They are either too complicated or has strange abstraction. The struct is very simple, no need to introduce another dependency.